src/meta/src/rpc/server.rs

```diff
@@ -339,6 +340,17 @@ pub async fn start_service_as_election_leader<S: MetaStore>(
     let system_params_manager = env.system_params_manager_ref();
     let system_params_reader = system_params_manager.get_params().await;
 
+    let data_directory = system_params_reader.data_directory();
+    if !is_correct_data_directory(data_directory) {
```

Comment:
IIUC a invalid data_directory won't be persisted (otherwise never get the chance to re-specify a correct one), because data_directory is persisted via `system_params_manager.flush_params` later. Could you verify it?

Reply:
Let me check it.